src/amqp_management.c

```diff
@@ -597,190 +600,252 @@ static int add_string_key_value_pair_to_map(AMQP_VALUE map, const char* key, con
+                                                {
+                                                    /* Codes_SRS_AMQP_MANAGEMENT_01_023: [ `amqp_management_create` shall create a message receiver by calling `messagereceiver_create` and passing to it the receiver link handle. ]*/
+                                                    amqp_management->message_receiver = messagereceiver_create(amqp_management->receiver_link, on_message_receiver_state_changed, amqp_management);
+                                                    if (amqp_management->message_receiver == NULL)
+                                                    {
+                                                        /* Codes_SRS_AMQP_MANAGEMENT_01_032: [ If `messagereceiver_create` fails then `amqp_management_create` shall fail and return NULL. ]*/
+                                                        LogError("Failed creating message receiver");
+                                                        link_destroy(amqp_management->receiver_link);
+                                                    }
+                                                    else
+                                                    {
+                                                        free(receiver_link_name);
+                                                        free(sender_link_name);
+                                                        amqpvalue_destroy(target);
+                                                        amqpvalue_destroy(source);
+
+                                                        /* Codes_SRS_AMQP_MANAGEMENT_01_106: [ The message Id set on the message properties shall start at 0. ]*/
+                                                        amqp_management->next_message_id = 0;
+
+                                                        goto all_ok;
+                                                    }
+
+                                                    messagesender_destroy(amqp_management->message_sender);
+                                                }
+
+                                                link_destroy(amqp_management->receiver_link);
                                             }
+
+                                            link_destroy(amqp_management->sender_link);
                                         }
+
+                                        free(receiver_link_name);
                                     }
+
+                                    free(sender_link_name);
                                 }
 
-                                free(receiver_link_name);
+                                amqpvalue_destroy(target);
                             }
 
-                            free(sender_link_name);
+                            amqpvalue_destroy(source);
                         }
 
-                        amqpvalue_destroy(target);
+                        free(amqp_management->status_description_key_name);
```

Comment:
>free(amqp_management->status_description_key_name); [](start = 24, length = 51)

Nit - along lines of other - could a single call in error case to amqp_management_destroy() be used here on those frees that are tied to the amqp_management struct?

Reply:
Same answer like for the other comment.

---
In reply to: [167072806](https://github.com/Azure/azure-uamqp-c/pull/211#discussion_r167072806) [](ancestors = 167072806)